### DISCUSSION TITLE
Keep cursor scrolloff position at end of file?

### DISCUSSION THREAD

**Discussion by mksonkar** [LazyVim/LazyVim#4827] [Q&A] [answered]
I've set `vim.opt.scrolloff = 10` so that there's minimum 10 lines below the cursor when I go down the file,
but when I reach the end of file, the scrolloff doesn't work and the cursor gets to the bottom. 

How can I prevent that? If I remember correctly there's was a way to show empty lines ~ at the bottom, but I don't know how.

**Comment by dpetka2001** [comment]
Something is overwriting the value of scrolloff (maybe a plugin you installed?). LazyVim has `scrolloff = 4` and it works perfectly fine for me.

**Reply by mksonkar** [reply]
I only have a couple of other plugins installed, and I just went through their configs. I don't seem to have any plugin that could be overriding the scrolloff option.

**Reply by mksonkar** [reply]
Can you recheck and confirm that for a larger file, the scrolloff remains when reaching end of file for you?

**Reply by dpetka2001** [reply]
Just try with default LazyVim installation without installing anything and see if that happens. It doesn't happen to me, so my guess (without being 100% sure) is that it's caused by some plugin conflict. What does `:verbose set scrolloff?` show?

**Reply by mksonkar** [reply]
```lua
:verbose set scrolloff?   
scrolloff=10
Last set from Lua (run Nvim with -V1 for more details)
```
I'll try with a fresh LazyVim config

**Reply by dpetka2001** [reply]
Oh I'm sorry!! I now see what you observe. It only happens when you get to the end of a file, right? My observations were somewhere inside the file but not at the end of it. Unfortunately I don't know why this happens or if there's another solution. 

Again, apologies for the confusion I created.

**Reply by mksonkar** [reply]
Its the same behaviour with a fresh LazyVim installation. 
The scrolloff works when going down with j, but for the last lines, the cursor goes to the bottom
![image](https://github.com/user-attachments/assets/3011bc4f-663f-445d-8bef-99e0139d3cd0)

**Reply by mksonkar** [reply]
> Oh I'm sorry!! I now see what you observe. It only happens when you get to the end of a file, right? My observations were somewhere inside the file but not at the end of it. Unfortunately I don't know why this happens or if there's another solution.
> 
> Again, apologies for the confusion I created.

It's okay. Maybe I could've described it better.

**Reply by dpetka2001** [reply]
Reading `:h 'scrolloff'` it mentions 
> except at the start or end of the file or when long lines wrap

So, it seems to be normal behavior? Still not aware if there's another workaround for it.

**Reply by dpetka2001** [reply]
When you're at the last line you can press `zz` to center the line on the screen. It's the only kind of work around I was able to find in Neovim help (although not the same as `scrolloff`). Unfortunately, I couldn't find anything else, but maybe somebody else can come up with a more proper solution.

**Reply by zvrkSam** [reply]
There is a plugin [Aasim-A/scrollEOF.nvim](https://github.com/Aasim-A/scrollEOF.nvim) I does the scrolloff at the end of the file.

**Comment by dimas-cyriaco** [comment]
It's not exactly what you are asking for, but I'm using this plugin to get something close to that: [rlychrisg/keepcursor.nvim](https://github.com/rlychrisg/keepcursor.nvim). It works mostly well, but, as other plugins that try to do this, it kind of breaks mouse scrolling.

**Reply by glarione** [reply]
tried this, doesn't work well with animations (trying to `gg` or `G` results in like 1 screen move to the top/bottom). 
But it works well with `git.blame`, without flickering in the git.blame buffer (scrolloff works bad with git.blame movement).
Is there any fix for mouse?

**Reply by sushi057** [reply, marked as answer]
Did you try scrollEOF.nvim?

**Reply by x3rAx** [reply]
> scrollEOF.nvim

It worked really great for a long time but recently my terminals started flickering / text scrolled all the way to the top. I had to set the option `disabled_filetypes = { "snacks_terminal" }` to fix it:

```
{
  "Aasim-A/scrollEOF.nvim",
  event = { "CursorMoved", "WinScrolled" },
  opts = {
    insert_mode = true,
    disabled_filetypes = {
      "snacks_terminal", -- Fix flickering in LazyGit and terminals
    },
  },
},
```
